cpr/session.cpp

```diff
@@ -97,6 +104,38 @@ Session::Impl::Impl() {
     }
 }
 
+void Session::Impl::SetReadCallback(const ReadCallback& read) {
+    CURL* curl = curl_->handle;
+    if (curl) {
+        curl_easy_setopt(curl, CURLOPT_READFUNCTION , read.callback.cb);
```

Comment:
I'm looking at this patch a little and noticed that CURLOPT_READFUNCTION is assigned a value of a type of `int(*read_cb_t)(ReadCallbackUser*);` (types in callback.h), whereas https://curl.haxx.se/libcurl/c/CURLOPT_READFUNCTION.html specifies a type of `size_t (*read_callback)(char *buffer, size_t size, size_t nitems, void *userdata);`.  It seems like that would cause a crash if used, due to the change: am I mistaken?